### PR TITLE
test initial support of python 3.12.x

### PR DESCRIPTION
**Pull request by auvipy**
To test and update python 3.12 support when it is released

### CODE REVIEW

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ['3.8', '3.9', '3.10', '3.11', 'pypy-3.9', 'pypy-3.10']
+        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12.0-rc.3' 'pypy-3.9', 'pypy-3.10']
```
Comment (auvipy) [inline .github/workflows/python-package.yml:32]:
```suggestion
        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12', 'pypy-3.9', 'pypy-3.10']
```

File: .github/workflows/python-package.yml [hunk 2]
```diff
@@ -29,7 +29,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ['3.8', '3.9', '3.10', '3.11', 'pypy-3.9', 'pypy-3.10']
+        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12', 'pypy-3.9', 'pypy-3.10']
```
Comment (auvipy) [inline .github/workflows/python-package.yml:32]:
```suggestion
        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12-dev', 'pypy-3.9', 'pypy-3.10']
```

**Review by auvipy — COMMENTED** [review]
=========================== short test summary info ============================
  FAILED t/unit/app/test_amqp.py::test_AMQP_proto1::test_as_task_message_without_utc
  FAILED t/unit/app/test_app.py::test_App::test_now - AttributeError: 'cached_p...
  FAILED t/unit/app/test_app.py::test_App::test_timezone__none_set - AttributeE...
  FAILED t/unit/app/test_app.py::test_App::test_uses_utc_timezone - AttributeEr...
  FAILED t/unit/app/test_app.py::test_App::test_compat_on_configure - Attribute...
  FAILED t/unit/app/test_app.py::test_App::test_send_task__connection_provided
  FAILED t/unit/app/test_app.py::test_App::test_select_queues - AttributeError:...
  FAILED t/unit/app/test_beat.py::test_schedule::test_maybe_make_aware - Attrib...
  FAILED t/unit/app/test_beat.py::test_schedule::test_to_local - AttributeError...
  FAILED t/unit/app/test_control.py::test_Control::test_after_fork_clears_mailbox_pool
  !!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 10 failures !!!!!!!!!!!!!!!!!!!!!!!!!!
  =========== 10 failed, 311 passed, 2 skipped, 120 warnings in 40.72s ===========
  3.12-unit: 1627010 C exit 1 (42.82 seconds) /home/runner/work/celery/celery> pytest --maxfail=10 --capture=no -v --cov=celery --cov-report=xml --cov-report term pid=7664 [tox/execute/api.py:279]

**Comment by auvipy** [comment]
Yes you are right! Will revisit this ASAP as it is a priority.

**Comment by auvipy** [comment]
AttributeError: 'cached_property' object has no attribute 'lock' is now fixed in kombu main branch

File: .github/workflows/python-package.yml [hunk 3]
```diff
@@ -29,7 +29,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ['3.8', '3.9', '3.10', '3.11', 'pypy-3.9', 'pypy-3.10']
+        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12-dev', 'pypy-3.9', 'pypy-3.10']
```
Comment (ThiefMaster) [inline .github/workflows/python-package.yml:32]:
```suggestion
        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12', 'pypy-3.9', 'pypy-3.10']
```

**Comment by eric-musliner** [comment]
What is the estimate for a release that supports 3.12. I see the fix merged in kombu already

**Pull request merged**